src/gui/fspathedit_p.cpp

```diff
@@ -80,6 +255,33 @@ void Private::FileLineEdit::keyPressEvent(QKeyEvent *e)
         m_completerModel->setRootPath(QFileInfo(text()).absoluteDir().absolutePath());
         showCompletionPopup();
     }
+
+    const FileSystemPathValidator *validator =
+        dynamic_cast<const FileSystemPathValidator *>(this->validator());
```

Comment:
Wouldn't it be better to add Q_OBJECT instead of using  another one rtti feature. Although, I'm not sure we never use it...